database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostViewModel.swift

```diff
@@ -62,6 +64,27 @@ class PostViewModel: ObservableObject, Identifiable {
     self.userIDsStarredBy = userIDsStarredBy
   }
 
+  func didTapSendButton(commentField: String) {
+    if let userID = Auth.auth().currentUser?.uid,
```

Comment:
Is it possible to tap the button when there isn't a current user? If so, should there then be a message or redirect to a login screen?

Reply:
The current logic is that the user needs to login in order to see the list of posts or click on posts. Since My Posts view and My Top Posts view needs to know who the current user is.

Reply:
In that case, will tapping the button silently fail if the user is not logged in?

Reply:
if the user is not logged in, they will only see the login/sign-up screen.

Reply:
Database will log errors to console, but otherwise yes. +1 to handling auth state errors more explicitly, but that can be done in a follow-up PR.